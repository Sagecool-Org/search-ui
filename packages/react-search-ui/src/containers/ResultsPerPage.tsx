import React, { Component } from "react";
import { withSearch } from "..";
import {
  ResultsPerPage,
  ResultsPerPageContainerContext,
  ResultsPerPageContainerProps,
  ResultsPerPageViewProps
} from "@elastic/react-search-ui-views";
import { useSearch } from "../hooks";

export function ResultsPerPageContainer({
  className,
  options,
  view,
  ...rest
}: ResultsPerPageContainerProps) {
  const {
    driver: {
      state: { resultsPerPage },
      actions: { setResultsPerPage }
    }
  } = useSearch();
  const View = view || ResultsPerPage;
  const viewProps: ResultsPerPageViewProps = {
    className,
    onChange: (value) => {
      setResultsPerPage(value);
    },
    options,
    value: resultsPerPage,
    ...rest
  };

  return <View {...viewProps} />;
}

export default ResultsPerPageContainer;
