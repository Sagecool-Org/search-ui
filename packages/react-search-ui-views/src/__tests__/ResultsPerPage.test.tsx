import React from "react";
import { render } from "@testing-library/react";
import ResultsPerPage from "../ResultsPerPage";

const requiredProps = {
  onChange: () => ({}),
  options: [20, 40]
};

it("renders correctly when there is a selected value", () => {
  const { container } = render(
    <ResultsPerPage {...requiredProps} value={40} />
  );
  expect(container).toMatchSnapshot();
});

it("renders correctly when there is not a selected value", () => {
  const { container } = render(<ResultsPerPage {...requiredProps} />);
  expect(container).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const { container } = render(
    <ResultsPerPage {...requiredProps} className={customClassName} />
  );
  expect(container.firstChild).toHaveClass(
    "sui-results-per-page",
    "test-class"
  );
});
