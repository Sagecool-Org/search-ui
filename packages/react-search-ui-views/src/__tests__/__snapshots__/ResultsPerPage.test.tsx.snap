// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is a selected value 1`] = `
<div>
  <div
    class="sui-results-per-page"
  >
    <div
      class="sui-results-per-page__label"
    >
      Show
    </div>
    <div
      class="sui-select sui-select--inline css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
        role="log"
      />
      <div
        class="sui-select__control css-0"
      >
        <div
          class="sui-select__value-container sui-select__value-container--has-value css-zfm7a7"
        >
          <div
            class="sui-select__single-value css-11aw737-singleValue"
          >
            40
          </div>
          <input
            aria-activedescendant=""
            aria-autocomplete="list"
            aria-expanded="false"
            aria-haspopup="true"
            aria-readonly="true"
            class="css-1s80ejz-dummyInput-DummyInput"
            id="react-select-2-input"
            inputmode="none"
            role="combobox"
            tabindex="0"
            value=""
          />
        </div>
        <div
          class="sui-select__indicators css-1wy0on6"
        >
          <span
            class="sui-select__indicator-separator css-0"
          />
          <div
            aria-hidden="true"
            class="sui-select__indicator sui-select__dropdown-indicator css-0"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when there is not a selected value 1`] = `
<div>
  <div
    class="sui-results-per-page"
  >
    <div
      class="sui-results-per-page__label"
    >
      Show
    </div>
    <div
      class="sui-select sui-select--inline css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
        role="log"
      />
      <div
        class="sui-select__control css-0"
      >
        <div
          class="sui-select__value-container css-zfm7a7"
        >
          <div
            class="sui-select__placeholder css-1jqq78o-placeholder"
            id="react-select-2-placeholder"
          >
            Select...
          </div>
          <input
            aria-activedescendant=""
            aria-autocomplete="list"
            aria-describedby="react-select-2-placeholder"
            aria-expanded="false"
            aria-haspopup="true"
            aria-readonly="true"
            class="css-1s80ejz-dummyInput-DummyInput"
            id="react-select-2-input"
            inputmode="none"
            role="combobox"
            tabindex="0"
            value=""
          />
        </div>
        <div
          class="sui-select__indicators css-1wy0on6"
        >
          <span
            class="sui-select__indicator-separator css-0"
          />
          <div
            aria-hidden="true"
            class="sui-select__indicator sui-select__dropdown-indicator css-0"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
